internal/controller/git/git.go

```diff
@@ -251,6 +274,30 @@ func (r *repo) CreateOrphanedBranch(branch string) error {
 	return r.Clean()
 }
 
+func (r *repo) CurrentBranch() (string, error) {
+	resBytes, err := libExec.Exec(r.buildCommand(
+		"branch",
+		"--show-current",
+	))
+	return strings.TrimSpace(string(resBytes)),
+		errors.Wrap(err, "error determining current branch")
+}
```

Comment:
All existing operations that change the current branch (`Checkout()`, for instance) update `repo.currentBranch`, so you can just return that.

Reply:
Actually, that was my first thought. But the reason this was necessary was because Checkout() is commonly called with an empty string and so `currentBranch` was being set to empty string.

If you'd like, I can improve checkout to also call `git branch --show-current` when empty string, and then this method becomes an accessor method to `currentBranch`.

Reply:
Oh... I forgot about that. If checkout is called with the empty string, you're just getting the trunk and you don't necessarily know what it's named.

> I can improve checkout to also call git branch --show-current when empty string

That seems like the right thing to do.